### PR TITLE
Fixes error when trying to upload GPS coordinate to Google Sheets

### LINKED ISSUE #819: Error when trying to submit to Google Sheets with a GPS coordinate
#### Software and hardware versions 
Collect v1.5.1-15-g3d6ffbb1, Android 6.0.1, Nexus 5

#### Problem description
Error when trying to submit to Google Sheets with a GPS coordinate. The exception is:

```
java.io.FileNotFoundException: /storage/emulated/0/odk/instances/demo-form_2017-04-03_23-39-07/55.5555555 -555.5555555 0.0 21.112: open failed: ENOENT (No such file or directory)
	at libcore.io.IoBridge.open(IoBridge.java:452)
	at java.io.FileInputStream.<init>(FileInputStream.java:76)
	at com.google.api.client.http.FileContent.getInputStream(FileContent.java:71)
	at com.google.api.client.googleapis.media.MediaHttpUploader.resumableUpload(MediaHttpUploader.java:397)
	at com.google.api.client.googleapis.media.MediaHttpUploader.upload(MediaHttpUploader.java:336)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:427)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
	at org.odk.collect.android.tasks.GoogleSheetsAbstractUploader.uploadFileToDrive(GoogleSheetsAbstractUploader.java:667)
	at org.odk.collect.android.tasks.GoogleSheetsAbstractUploader.uploadOneSubmission(GoogleSheetsAbstractUploader.java:324)
	at org.odk.collect.android.tasks.GoogleSheetsAbstractUploader.uploadInstances(GoogleSheetsAbstractUploader.java:130)
	at org.odk.collect.android.activities.GoogleSheetsUploaderActivity$GoogleSheetsInstanceUploaderTask.doInBackground(GoogleSheetsUploaderActivity.java:555)
	at org.odk.collect.android.activities.GoogleSheetsUploaderActivity$GoogleSheetsInstanceUploaderTask.doInBackground(GoogleSheetsUploaderActivity.java:518)
	at android.os.AsyncTask$2.call(AsyncTask.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
	at libcore.io.Posix.open(Native Method)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:186)
	at libcore.io.IoBridge.open(IoBridge.java:438)
```

#### Steps to reproduce the problem
1. Change protocol to "Google Sheets, Google Drive"
1. Get a form with a GPS coordinate such as the one described [here](https://nafundi.com/blog/posts/serverless-odk-deployments-with-google-apps/)
1. Fill out a GPS coordinate
1. Finalize and submit the form

#### Other information 
It looks like the coordinates are being treated as a filename. This is likely related to #780 so @shobhitagarwal1612 you may want to take a quick look since you know that code well!

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #819

### CODE REVIEW

**Comment by shobhitagarwal1612** [comment]
@lognaturel Please review

**Comment by lognaturel** [comment]
@shobhitagarwal1612 You're welcome to tag the reviewers if you're not getting a review within 10 days but within that time frame, we are doing our best to review when we can!

In this case, my initial reaction is that I don't think this is the right approach. I want to give myself some more time to think about it and possibly offer an alternative, however.

**Comment by shobhitagarwal1612** [comment]
Sorry for being so impatient.
It was done in a hurry as I thought it needed an urgent fix so I did not think whether it's the best approach or not. It would be nice to have some opinion regarding a better solution.
Currently what I have done is check whether string matches the pattern for filename. I don't think so that the file formats that have to be uploaded have a full numeric extension but if the filename is a gps coordinate, then that would be a complete numeric.

**Comment by lognaturel** [comment]
@shobhitagarwal1612 Unless we're somehow corrupting data or preventing use of a feature, we usually just wait until the next release. We'd make it super clear if something is urgent in some way. 😊

What I'm more worried about is that if my name is "foo.jpg", this fails. I wanted to double check it but had some trouble sending to sheets when I first looked at it. I think there has been some instability to that infrastructure recently. Next time I should just tell you what I think the problem is and you can look into it!

I think it would be good to look at the aggregate upload task and see how it determines what fields are files to send.

**Comment by shobhitagarwal1612** [comment]
That is a great idea. I'll check how it is being done in aggregate.
> What I'm more worried about is that if my name is "foo.jpg", this fails.

I have tested uploading all kinds of file types and it seemed to work for me. It was also able to distinguish GPS coordinate.

**Comment by lognaturel** [comment]
Yes, but what if you enter a string with a .* pattern?? It tries to upload that and shows an error because it can't find it. A more realistic example might be that someone is filling out a report: "blah blah blah. And I also noticed...". No text field can have periods because they will be considered filenames. The bug is broader than just GPS coordinates.

**Comment by shobhitagarwal1612** [comment]
Using regex to check whether it's a file or not was a wrong choice. 
Now, i'm checking whether the path points to a file or not.

screenshot_2017-04-06_15-29-37
screenshot_2017-04-06_15-27-28

P.S. I couldn't figure out the way aggregate does uploading.  
Please check if this is the right approach

File: collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java [hunk 1]
```diff
@@ -813,13 +813,13 @@ private void processInstanceXML(File instanceFile,
...
         parser.setInput(in, null);
-        readInstanceFeed(parser, answersToUpload, mediaToUpload);
+        readInstanceFeed(parser, answersToUpload, mediaToUpload, instanceFile);
         in.close();
     }
 
     private void readInstanceFeed(XmlPullParser parser,
                                   HashMap<String, String> answersToUpload,
-                                  HashMap<String, String> mediaToUpload)
+                                  HashMap<String, String> mediaToUpload, File instanceFile)
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java]: Here what you really need is the instance folder path, right? I'd pass in something like `File instanceFolder` and just call `getParentFile()` once when you make the call. That makes the intent a little bit clearer I think.

**Review by lognaturel — CHANGES_REQUESTED** [review]
Yay, this works great. 😊 Small suggestion and then we're ready to go.

**Pull request merged**